### PR TITLE
use bin/php when composer is found

### PR DESCRIPTION
**Pull request by lewactwo**
| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

### CODE REVIEW

**Comment by elfet** [comment]
Composer itself is a bin and can be called directly.

**Comment by mbrodala** [comment]
@elfet True, but it will use `/usr/bin/env php` which is not the correct PHP CLI version to run on all servers, thus I consider this PR very valid.

**Comment by lewactwo** [comment]
@elfet Yes it can, but on my hosting I have more php versions - `php55`, `php56`, `php70` and `php71`. Default php binary is symlink to `php56`. In config I set `bin/php` to `php71` and it works for everything but not for composer. Some of my packages require PHP 7.0 and without that I got error. I think if somebody set `bin/php` he want to use that php binary everywhere.

File: recipe/common.php [hunk 1]
```diff
@@ -91,7 +91,7 @@
 
 set('bin/composer', function () {
     if (commandExist('composer')) {
-        $composer = run('which composer')->toString();
+        $composer = '{{bin/php}} ' . run('which composer')->toString();
```
Comment (elfet) [inline recipe/common.php]: Agree.

Comment (pluseg) [inline recipe/common.php]: @lewactwo Kamil, could you please update your PR with @mbrodala comment? It's about little duplication of `{{bin/php}} ` part.

Comment (lewactwo) [inline recipe/common.php]: @pluseg It's done.

**Pull request merged**